core/src/introspection.cpp

```diff
@@ -64,34 +64,74 @@ std::string getTaskId(const TaskPrivate* task) {
 }
 }  // namespace
 
+/**
+ * A shared executor between all tasks' introspection, this class will keep track of the number of the added nodes and
```

Comment:
@henningkayser

Reply:
This executor is only really used for the service, right? What's the implication of simply reusing the same node and just appending the private task namespace to the all topics? I'm pretty sure your approach works as expected, I'm just not sure if it's really necessary.